### PR TITLE
add solution to kata 3 get/api

### PR DESCRIPTION
**Pull request by milosh1982**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -32,3 +32,17 @@ describe("GET api/nonsense ", () => {
       });
   });
 });
+describe("GET /api", () => {
+  test("should respond with an object describing all the available endpoints ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const { allApi } = body;
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: I would consider renaming this to endpoints rather than allApi

File: __model__/api.model.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const db = require("../db/connection");
+const fs = require("fs/promises");
+
+exports.selectApi = () => {
+  return fs.readFile(`${__dirname}/../endpoints.json`, "utf-8").then((data) => {
+    const parsedData = JSON.parse(data);
+    return parsedData;
+  });
+};
```
Comment (Hard-Luck) [inline __model__/api.model.js]: This is not interacting with the database so isn't strictly speaking a model, consider making this a util function if you want the logic separate from the controller

File: __tests__/app.test.js [hunk 3]
```diff
@@ -32,3 +32,17 @@ describe("GET api/nonsense ", () => {
...
+        const arrOfKeys = Object.keys(allApi);
+        arrOfKeys.forEach((key) => {
+          expect(typeof key).toBe("string");
+        });
```
Comment (Hard-Luck) [inline __tests__/app.test.js:47]: Every key on an object is a string, you could be more specific with your test, you know exactly what the response should look like.

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Just some small changes to be made, along with what I've commented you should change your folder names, models and controllers should have __either side__

after the changes, resubmit the PR

File: __tests__/app.test.js (hunk 3 above)
Comment (milosh1982) [reply __tests__/app.test.js:47]: The problem I have that response should be dynamic. So don't want to use toEqual. And really don't know how to make it dynamic.

Comment (Hard-Luck) [reply __tests__/app.test.js:47]: You could read the file contents and compare the output to that couldn't you, both of those would hopefully always be the same.

**Review by Hard-Luck — APPROVED** [review]
Potential small change then merge

**Pull request merged**